### PR TITLE
Waagent

### PR DESCRIPTION
**Pull request by pdostal**
- Related ticket: [poo#162950](https://progress.opensuse.org/issues/162950)
- Verification runs:
[Build20240701-1](https://pdostal-server.suse.cz/tests/overview?build=20240701-1&distri=sle&version=15-SP6) of sle-15-SP6-EC2-Updates.aarch64	  [publiccloud_consoletests@64bit](https://pdostal-server.suse.cz/tests/6686)	[24](https://pdostal-server.suse.cz/tests/6686)	3 minutes ago
[Build20240702-1](https://pdostal-server.suse.cz/tests/overview?build=20240702-1&distri=sle&version=15-SP6) of sle-15-SP6-Azure-BYOS-Updates.aarch64	  [publiccloud_consoletests@64bit](https://pdostal-server.suse.cz/tests/6683)	[24](https://pdostal-server.suse.cz/tests/6683)	20 minutes ago
[Build0611](https://pdostal-server.suse.cz/tests/overview?build=0611&distri=sle&version=15-SP6) of sle-15-SP6-EC2.x86_64	  [publiccloud_consoletests@ec2_t2.large](https://pdostal-server.suse.cz/tests/6685)	[20](https://pdostal-server.suse.cz/tests/6685)	22 minutes ago
[Build20240701-1](https://pdostal-server.suse.cz/tests/overview?build=20240701-1&distri=sle&version=15-SP6) of sle-15-SP6-GCE-BYOS-Updates.x86_64	  [publiccloud_consoletests@64bit](https://pdostal-server.suse.cz/tests/6687)	[24](https://pdostal-server.suse.cz/tests/6687)	23 minutes ago
[Build0603](https://pdostal-server.suse.cz/tests/overview?build=0603&distri=sle&version=15-SP6) of sle-15-SP6-Azure-BYOS.aarch64	  [publiccloud_consoletests@64bit](https://pdostal-server.suse.cz/tests/6682)	[20](https://pdostal-server.suse.cz/tests/6682)	28 minutes ago
[Build0603](https://pdostal-server.suse.cz/tests/overview?build=0603&distri=sle&version=15-SP6) of sle-15-SP6-Azure-BYOS.aarch64	  [publiccloud_consoletests@64bit](https://pdostal-server.suse.cz/tests/6688)	[20](https://pdostal-server.suse.cz/tests/6688)	29 minutes ago
[Build0568](https://pdostal-server.suse.cz/tests/overview?build=0568&distri=sle&version=15-SP6) of sle-15-SP6-GCE.x86_64	  [publiccloud_consoletests@gce_n1_standard_2](https://pdostal-server.suse.cz/tests/6684)	[20](https://pdostal-server.suse.cz/tests/6684)	30 minutes ago
[Build0043](https://pdostal-server.suse.cz/tests/overview?build=0043&distri=sle-micro&version=6.0) of sle-micro-6.0-Azure-BYOS.aarch64	  [publiccloud_upload_img@64bit](https://pdostal-server.suse.cz/tests/6667)	[2](https://pdostal-server.suse.cz/tests/6667)	about 3 hours ago

### CODE REVIEW

**Review by ricardobranco777 — APPROVED** [review]
I notice that at least one service in the poo# ticket is missing:
- cloud-regionsrv-client-plugin-ec2

Otherise LGTM

**Review by grisu48 — APPROVED** [review]
LGTM!

**Comment by grisu48** [comment]
Careful though: Let's either exclude this new module for SLEM on PC, or do VRs for SLEM on PC. I'm not sure if e.g. `cloud-init` is part of SLEM on Azure/EC2.

**Comment by asmorodskyi** [comment]
I think we don't want to increase diff between maintenance and latest but decrease it . Can you please also add this scenario to latest ?

File: tests/publiccloud/check_services.pm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    # cloud-init
+    if (is_azure || is_ec2) {
+        assert_script_run('systemctl status cloud-init.service');
+        assert_script_run('systemctl status cloud-init.target');
+        assert_script_run('systemctl status cloud-init-local.service');
+    }
+
+    # cloud-config
+    if (is_azure || is_azure) {
```
Comment (asmorodskyi) [inline tests/publiccloud/check_services.pm:39]: I would assume this is typo :thinking:

File: tests/publiccloud/check_services.pm [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        assert_script_run('systemctl status cloud-config.service');
+        assert_script_run('systemctl status cloud-config.target');
+    }
+
+    # google-guest-agent
+    if (is_gce) {
```
Comment (asmorodskyi) [inline tests/publiccloud/check_services.pm:45]: can we merge this if and one below ?

**Comment by pdostal** [comment]
> Careful though: Let's either exclude this new module for SLEM on PC, or do VRs for SLEM on PC. I'm not sure if e.g. `cloud-init` is part of SLEM on Azure/EC2.

I don't see where this would be executed on SLEM. I see only LTP and SLEM specific tests. No consoletests.

**Comment by asmorodskyi** [comment]
> > Careful though: Let's either exclude this new module for SLEM on PC, or do VRs for SLEM on PC. I'm not sure if e.g. `cloud-init` is part of SLEM on Azure/EC2.
> 
> I don't see where this would be executed on SLEM. I see only LTP and SLEM specific tests. No consoletests.

you added line **before**  ` if (get_var('PUBLIC_CLOUD_CONSOLE_TESTS')) {`  so this module will be added not only to console tests but also to : 
1. containers tests 
2. xfs tests 
3. smoke tests 
4. nfs tests 
5. nvidia tests

**Comment by asmorodskyi** [comment]
> > > Careful though: Let's either exclude this new module for SLEM on PC, or do VRs for SLEM on PC. I'm not sure if e.g. `cloud-init` is part of SLEM on Azure/EC2.
> > 
> > 
> > I don't see where this would be executed on SLEM. I see only LTP and SLEM specific tests. No consoletests.
> 
> you added line **before** ` if (get_var('PUBLIC_CLOUD_CONSOLE_TESTS')) {` so this module will be added not only to console tests but also to :
> 
> 1. containers tests
> 2. xfs tests
> 3. smoke tests
> 4. nfs tests
> 5. nvidia tests

but yeah no SLEM tests they are triggered via different `main_micro_alp.pm`

**Comment by pdostal** [comment]
@asmorodskyi let's move it to console tests then?

File: tests/publiccloud/check_services.pm (hunk 2 above)
Comment (pdostal) [reply tests/publiccloud/check_services.pm:45]: Done.

File: tests/publiccloud/check_services.pm (hunk 1 above)
Comment (pdostal) [reply tests/publiccloud/check_services.pm]: Done.

**Comment by pdostal** [comment]
> * cloud-regionsrv-client-plugin-ec2

I didn't find it later on :) :( I hope it's OK if I skip it for now.

**Review by ricardobranco777 — APPROVED** [review]

**Pull request merged**